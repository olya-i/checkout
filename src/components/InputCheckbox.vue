<script lang="ts">
import { defineComponent, type PropType } from "vue";

export default defineComponent({
  props: {
    label: String,
    name: String,
    modelValue: Boolean as PropType<unknown>,
    isInvalid: Boolean,
  },
  emits: ["update:modelValue", "validate"],
  computed: {
    value: {
      get() {
        return this.modelValue as boolean;
      },
      set(value: boolean) {
        this.$emit("update:modelValue", value);
        this.$emit("validate");
      },
    },
  },
});
</script>

<template>
  <div class="checkbox-wrapper">
    <label class="label" :class="{ error: isInvalid }">
      <input type="checkbox" :name="name" v-model="value" />
      {{ label }}
    </label>
  </div>
</template>

<style lang="scss" scoped>
.checkbox-wrapper {
  margin-bottom: 20px;
}
.label {
  font-size: 20px;

  &.error {
    color: var(--red);
  }
}
</style>
