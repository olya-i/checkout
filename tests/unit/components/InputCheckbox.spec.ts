import InputCheckbox from "../../../src/components/InputCheckbox.vue";
import { mount } from "@vue/test-utils";

describe("Mounted InputCheckbox", () => {
  it("renders correctly", () => {
    const wrapper = mount(InputCheckbox, {
      props: {
        label: "Checkbox label",
        name: "checkbox",
        modelValue: false,
        isInvalid: false,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
