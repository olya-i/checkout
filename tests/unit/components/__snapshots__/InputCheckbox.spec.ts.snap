// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mounted InputCheckbox renders correctly 1`] = `
<div
  class="checkbox-wrapper"
>
  <label
    class="label"
  >
    <input
      name="checkbox"
      type="checkbox"
    />
     Checkbox label
  </label>
</div>
`;
